cosmos-proxy/src/cache.js

```diff
@@ -36,17 +36,22 @@ function isCacheEmpty() {
 function isCacheAuthenticated(reqUser, token) {
     if (!isCacheEmpty()) {
         for (var i = 0; i < cache.length; i++) {
-            if ((cache[i]['user'] === reqUser) && (cache[i]['token'] === token)) {
-                return true;
+            if (cache[i].user === reqUser) {
+                if (cache[i].token === token) {
+                    return 1; // 1
+                } else {
+                    return 2; // 2
+                } // if else
             } // if
         } // for
-    } // if
-
-    return false;
+        return 0;
+    } else {
+        return 0;
+    } // if else
 } // isCacheAuthenticated
 
 function updateCacheFile() {
-    fs.writeFileSync(pathToFile,'[' + cache + ']');
+    fs.writeFileSync(pathToFile, JSON.stringify(cache), 'utf-8');
```

Comment:
It's `utf8`, you're right. Checked in official page https://nodejs.org/api/fs.html#fs_fs_writefilesync_file_data_options :) 
Fixed in  bcabf88